### DISCUSSION TITLE
Estimate a single beta per condition instead of one beta per stimulus presentation

### DISCUSSION THREAD

**Discussion by alireza-kr** [cvnlab/GLMsingle#134] [Q&A] [answered]
Hello

I am using Python and want to use GLMsingle for my experiment. I aim to estimate a single beta for each condition in my experiment instead of one beta per stimulus presentation, as GLMsingle does. Since I have multiple instances of the same condition in my experiment, I believe the matrix with multiple observations per column is better conditioned than the matrix with only one observation per column.
I know that GLMdenoise does this, but there is no Python implementation except [pyGLMdenoise](https://github.com/Charestlab/pyGLMdenoise) by @iancharest. I was wondering if it is possible to force GLMsingle to estimate a single beta per condition rather than one beta per stimulus presentation.

Thanks

**Comment by kendrickkay** [comment, marked as answer]
I understand your point. The architecture of GLMsingle is not easily adaptable to "forcing" it to estimate one beta per condition. The easiest thing to do is to just post-hoc average the trials corresponding to your condition of interest.

**Comment by alireza-kr** [comment]
I was wondering how I can modify GLMsingle to estimate a single beta for each condition instead of one beta per stimulus presentation. In 'glmsingle.py', I found a variable called 'designSINGLE', which I believe is the single-trial version of the design matrix. Would it be possible to achieve the desired output by simply setting 'designSINGLE = design' in 'glmsingle.py'?

**Comment by kendrickkay** [comment]
No, it is not as simple as that...   The algorithm is more involved/complicated.  In order to perform cross-validation, the architecture is currently designed to cross-validate beta estimates across runs (and these beta estimates are fundamentally single-trial-oriented).

**Comment by iancharest** [comment]
Also, to add to what Kendrick mentioned, there are several benefits in GLMsingle that may outweigh our previous GLMdenoise. For example, beta weights are optimised with a family of HRFs. Have you tried running GLMsingle, and averaging across repetitions?

**Comment by alireza-kr** [comment]
Thank you for your answers. I am trying to average across repetitions. I haven't compared the results with GLMdenoise or first-level analysis in Nilearn yet, but I noticed a [discussion](https://github.com/cvnlab/GLMsingle/discussions/122) by Aakash where he compared results from GLMsingle with GLMdenoise and SPM.
